components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/service/DeviceOrganizationProviderServiceImpl.java

```diff
@@ -0,0 +1,331 @@
+        } catch (DeviceOrganizationException e) {
+            String msg = "Error while getting devices in organization";
+            log.error(msg, e);
+        }
+        List<DeviceOrganizationVisNode> nodes = new ArrayList<>();
+        for (DeviceOrganizationMetadataHolder tempHolder: tempDevicesInOrganization) {
+            String tempId = tempHolder.getDeviceId();
+            String tempLabel = tempHolder.getDeviceName();
+            int tempSize = 21;
+            String tempColor = "#0000ff";
+            if (tempHolder.getDeviceId().equals("server")) {
+                tempColor = "#ffa500";
+                tempSize = 70;
+            }
+            if (!tempHolder.getDeviceId().equals("server") && tempHolder.getIsGateway() == 1) {
+                tempSize = 33;
+                tempColor = "#ff0000";
+            }
+            if (tempHolder.getState() == 1 || tempHolder.getState() == 0) {
+                tempColor = "#d3d3d3";
+            }
+            nodes.add(new DeviceOrganizationVisNode(tempId,tempLabel,tempSize,tempColor));
+        }
+        return nodes;
+    }
+
+    @Override
+    public List<DeviceOrganizationVisEdge> generateEdges() {
+        List<DeviceOrganizationMetadataHolder> tempDevicesInOrganization = new ArrayList<>();
+        try {
+            tempDevicesInOrganization = this.getDevicesInOrganization();
+        } catch (DeviceOrganizationException e) {
+            String msg = "Error while getting devices in organization";
+            log.error(msg, e);
+        }
+        List<DeviceOrganizationVisEdge> edges = new ArrayList<>();
+        for (DeviceOrganizationMetadataHolder tempHolder: tempDevicesInOrganization) {
+            String child = tempHolder.getDeviceId();
+            String parent = tempHolder.getParent();
+            if (parent.equals("") || parent.equals(" ")) {
+                continue;
+            } else {
+                edges.add(new DeviceOrganizationVisEdge(parent, child));
+            }
+        }
+        return edges;
+    }
+
+    @Override
+    public List<DeviceOrganizationNode> generateHierarchy() {
+        List<DeviceOrganizationNode> hierarchy = new ArrayList<>();
+        List<DeviceOrganizationNode> parentList = new ArrayList<>();
+        List<DeviceOrganizationNode> childList = new ArrayList<>();
+        hierarchy.add(new DeviceOrganizationNode("server"));
+        parentList.add(new DeviceOrganizationNode("server"));
+        try {
+            childList = transformMetadataHolderArray(deviceOrganizationDAOimpl.getChildrenByParentId("server"));
+        } catch (DeviceOrganizationDAOException e) {
+            String msg = "Error while retrieving immediate children of server";
+            log.error(msg, e);
+        }
+        int count = 0;
+        do {
+            parentList.clear();
+            parentList.addAll(childList);
+            if (count == 0) {
+                hierarchy.get(0).setChildren(childList);
+            } else {
+
```

Comment:
why empty `else` block here

Reply:
This code is for an unimplemented method for a propsed feature 83efdcea4c6b6d0595c9334f76f4529ec14a5581. It has been removed.